langchain-core/src/utils/testing/index.ts

```diff
@@ -212,30 +214,96 @@ export class FakeChatModel extends BaseChatModel {
   }
 }
 
-export class FakeStreamingChatModel extends BaseChatModel {
-  sleep?: number = 50;
+export class FakeStreamingChatModel extends BaseChatModel<FakeStreamingChatModelCallOptions> {
+  sleep: number = 50;
 
-  responses?: BaseMessage[];
+  responses: BaseMessage[] = [];
+
+  chunks: AIMessageChunk[] = [];
+
+  toolStyle: "openai" | "anthropic" | "bedrock" | "google" = "openai";
 
   thrownErrorString?: string;
 
-  constructor(
-    fields: {
-      sleep?: number;
-      responses?: BaseMessage[];
-      thrownErrorString?: string;
-    } & BaseLLMParams
-  ) {
-    super(fields);
-    this.sleep = fields.sleep ?? this.sleep;
-    this.responses = fields.responses;
-    this.thrownErrorString = fields.thrownErrorString;
+  private tools: StructuredTool[] = [];
```

Comment:
should depend on the interface here

Reply:
Done. tools is now typed as (StructuredTool | ToolSpec)[], so we accept both full StructuredTool instances and lightweight { name, description, schema } objects.

@benjamincburns 

Let me know if you’d like any further tweaks/changes!